### PR TITLE
Enhanced gemma prediction with new flawless logit

### PR DESCRIPTION
**Pull request by carlofisicaro**
Integration of a `flawless_logit`

- Derived by first computing logits for each token in isolation, normalizing these logits, and then subtracting the normalized sum from the baseline_logits.
- Encoding each token individually.
- Applying a normalization step (final_norm).
- Decoding the normalized tokens to get logits.
- Normalizing these logits using a softmax function.
- Summing the normalized logits.
- Subtracting this sum from the gemma2 original logit

By normalizing the logits for each token, the model ensures that the predictions are more balanced and less likely to be dominated by any single token.
Subtracting the normalized sum can help reduce biases and make the logits more representative of the actual distribution of the data.

Based on initial tests conducted on Gemma2 7B, it appears that the performance at inference time has been improved.

### CODE REVIEW

**Comment by carlofisicaro** [comment]
The GitHub CLA check doesn't recognize the noreply user @a-googler <no****ly​@google.com>.

How shall I proceed? Should I use an interactive rebase to edit the author of the related commits?
